Database/MongoDB/Transport/Tls.hs

```diff
@@ -45,15 +48,20 @@ import Database.MongoDB.Query (access, slaveOk, retrieveServerData)
 
 -- | Connect to mongodb using TLS
 connect :: HostName -> PortID -> IO Pipe
-connect host port = bracketOnError (connectTo host port) hClose $ \handle -> do
+connect = connectWithTlsParams Nothing
+
+-- | Connect to mongodb using TLS using provided TLS client parameters
+connectWithTlsParams :: Maybe TLS.ClientParams -> HostName -> PortID -> IO Pipe
+connectWithTlsParams clientParams host port = bracketOnError (connectTo host port) hClose $ \handle -> do
```

Comment:
I'd suggest to make `clientParams` mandatory and move `fromMaybe params clientParams` thingy to `connect`. Two reasons:
- the functions is called `withParams`, it makes little sense not to pass params, one can use `connect` if necessary;
- the default params are bad (the skip certificate check) and should not be used, so it's better to plan to deprecate `connect` instead of adding another function with bad defaults
Just my 2c

Reply:
Sure. I updated the PR with the suggested changes.